README.md

```diff
@@ -9,10 +9,60 @@ In this project we use:
  - [`SPARK`](https://learn.adacore.com/courses/intro-to-spark/index.html) and [`GNATprove`](https://github.com/AdaCore/spark2014) for formal verification
+ - [`GNAT`](https://gcc.gnu.org/onlinedocs/gcc-12.1.0/gnat_ugn/Warning-Message-Control.html#Warning-Message-Control) built-in checks for static analysis
  - [`AUnit`](https://github.com/AdaCore/aunit) for unit-testing
  - [`GNATcoverage`](https://github.com/AdaCore/gnatcoverage) for source code coverage analysis
  - [`GitHub Actions`](https://docs.github.com/en/actions) for Continuous Integration
 
+# SPARK proof
+
+Automatic formal verification of the [SPARK
+code](https://www.adacore.com/about-spark) in this project is done by the
+GNATprove tool. This tool is run as the [Continuous
+Integration](#continuous-integration-github-actions) process to ensure that
+contributions do not break the proof.
+
+# Static Analysis
+
+The GNAT compiler comes with a lot of static analysis checks built-in. When
+using the `validation` [build profile](#build-profiles), Alire enables all the
+static analysis checks of GNAT. This provides detection of dead code, unused
+variables, or uninitialised data for instance.
+
+# Unit testing
+
+A local Alire crate in the `tests/` folder is used to separate the test
+code/framework from the main project. Using a local crate also means that we
+can have extra dependencies only for testing:
+ - Unit test framework
+ - Libraries to read test data (TOML, JSON, etc.)
+ - GNATcoverage for [code coverage analysis](#code-coverage)
+ - Etc.
+
+This project uses [AUnit](https://github.com/AdaCore/aunit) as unit testing
+framework. The `tests/` crate is both built and run as part of the [Continuous
+Integration](#continuous-integration-github-actions) process.
+
+# Code Coverage
+
+GNATcov is used to for code coverage analysis on execution of the
+[tests](#unit-testing). As part of the [Continuous
+Integration](#continuous-integration-github-actions) process, the coverage
+analysis is automatically produced and uploaded to an online service called
+[codecov.io](https://codecov.io).
+
+With Codecovio you can also have a nice "badge" to display the latest coverage
+analysis result in your GitHub `readme.md` file:
+[![codecov](https://codecov.io/gh/alire-project/ada_spark_workflow/branch/main/graph/badge.svg?token=9PZQ67LBPF)](https://codecov.io/gh/alire-project/ada_spark_workflow)
+
+# Continuous Integration (GitHub Actions)
+
+For this project we use GitHub Actions for Continuous Integration (CI).
+It provides virtual machines to build and test the project when new code is added, as well as automated validation of contributions through pull-request checks, or code coverage report with codecovio.
+
+The only thing needed for CI to work is a "workflow" file in the `.github/workflows/` directory.
+
+We commented as much as possible the workflow file of this repository so if you want to know more about how it works please have a look [here](.github/workflows/main.yml).
```

Comment:
Regarding "We commented as much as possible the workflow file", this is a minor thing but a native English speaker wouldn't use that word order.
Maybe "We extensively commented the workflow file of this repository so if..."